FitGoal/FitGoal/HomeViewController/CollectionViewSectionsAndHeaders/RoutineInspectorCell.swift

```diff
@@ -26,9 +30,27 @@ class RoutineInspectorCell: UICollectionViewCell {
         routineCollectionView.delegate = self
         routineCollectionView.dataSource = self
         routineCollectionView.register(ExerciseCell.self, forCellWithReuseIdentifier: ExerciseCell.identifier)
+        routineCollectionView.isUserInteractionEnabled = true
         
         let layout = routineCollectionView.collectionViewLayout as! UICollectionViewFlowLayout
         layout.scrollDirection = .horizontal
+        
+        let tap = UITapGestureRecognizer(target: self, action: #selector(handleTap(_:)))
+        self.addGestureRecognizer(tap)
+    }
+    
+    @objc func handleTap(_ sender: UITapGestureRecognizer) {
+        
+        switch sender.state {
+        case .ended:
+            let tapLocation = sender.location(in: routineCollectionView)
+            guard let indexPath = routineCollectionView.indexPathForItem(at: tapLocation) else { return }
+            guard let cell = routineCollectionView.cellForItem(at: indexPath) as? ExerciseCell else { return }
+            guard let exercise = cell.exercise else { return }
+            routineInspectorCellDelegate?.userDidSelectExersice(exercise)
+        default:
+            return
+        }
```

Comment:
You don't need a custom way of detecting the selection in a collection view. To do that you should implement the function: 

```swift
func collectionView(UICollectionView, didSelectItemAt: IndexPath)
```

of the `UICollectionViewDelegate` protocol.